### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by jerivas**
## Description
Use GitHub Actions to lint, test, and check coverage of the codebase.

## Steps to test/reproduce
This pull request should now include GH Actions checks at the bottom. Once this is merged the checks will run on all `push` events for all branches.

## Show me
New GitHub Actions checks are passing

### CODE REVIEW

**Comment by jerivas** [comment]
[Migrate from Travis to GH Actions on True and Accoutrement.](https://trello.com/c/3A4TzAMM/66-migrate-from-travis-to-gh-actions-on-true-and-accoutrement)

**Comment by jerivas** [comment]
@jgerigmeyer from my testing it looks like [coveralls](https://coveralls.io/github/oddbird/true?branch=gh-actions) is still working as expected. Not sure if you want to keep it or switch to codecov

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  push:
+
+jobs:
+  test:
+    name: Node ${{ matrix.node }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node: ['12', '14', '16']
```
Comment (jgerigmeyer) [inline .github/workflows/test.yml:13]: Ideally I'd like to test against the oldest version we support (12), the LTS version, and the latest version. The [docs](https://github.com/actions/setup-node#supported-version-syntax) make it sound like maybe they support this nvm syntax?

```suggestion
        node: ['12', 'lts/*', 'node']
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+          cache: yarn
+      - run: yarn install
```
Comment (jgerigmeyer) [inline .github/workflows/test.yml:20]:
```suggestion
      - run: yarn install --frozen-lockfile
```

**Review by jgerigmeyer — APPROVED** [review]
Looks good! I'm torn on coveralls/codecov -- honestly with our workflow I'm not sure how much value they add.

**Pull request merged**